blog-posts/event-sourcing-serveless/event-sourcing-serveless.md

```diff
@@ -0,0 +1,166 @@
+- You can only plug **2 lambda maximum** to the streams, and generally application contain more than 2 sets of data to display therefore more than 2 projectors.
+
+The fanout delivers events to one or multiple recipients. By doing so, each projector only needs to listen to events altering the data it's associated to. For example a projector writing user individual information such as the name family name and address doesn't need to be triggered by a credit event.
+
+![Archi without dispatch aggregate](./assets/archiWoDispatchAggregate.png 'Archi without dispatch aggregate')
+
+## From a dev reliable architecture to a prod-safe architecture
+
+This works fine in dev but **not in production yet**. This data synchronization architecture will face bugs later on. Let me explain :
+
+- When traffic starts to grow, more and more events will flow and one thing **EventBridge doesn't ensure** is the **order of the events** sent in the bus event. In order not to overwrite an information with an older one, the last version of event that updated the projection must be kept track of.
+- EventBridge ensures that an **event** is sent at least once, but it can still be **received more than once**. So projections need to be idempotent.
+- In dev, when a **projection** needs to be **updated** or a new one needs to be created, wiping the databases and creating new data is the easy way to go to test this new feature. But in production that is not an option. Projections are made from the succession of all events that happened, that's why the solution is to find a way to **replay** all events to recreate the projections. Projectors are idempotent and disorder-proof, so they can handle smoothly a replay of all events.
+
+## :unamused: EventBridge Archive, the bogus good idea
+
+> :bulb: EventBridge offers the possibility to replay all events that went through its service. This feature is called **EventBridge Archive**.
+
+There are [2 main issues with that solution](#whats-left-for-me-to-do-to-implement-the-most-effective-solution-with-the-latest-releases):
+
+- actually Archive is **not 100% accurate** and might be incomplete or have a bit more events than what's stored in the event ledger
+- replaying all events implies a **high traffic** when having a high number of events
+
+> :star2: If you think about it, none of the above problems would be one if only projectors had access to the aggregate when updating a projection.
+
+## :magic_wand: A trick to make data synchronization and replay easier
+
+The trick I came up with is implementing **state carried events**. Instead of computing the same aggregate in every projector and giving them all access to the event ledger, the **aggregate** is **computed once** and it is **attached to the dispatched events**. It takes the form of a new Lambda function that takes only one event in argument, computes the aggregate and attaches it to the event before republishing it. I named it `dispatchAggregate`.
```

Comment:
"I came up with" ? Tu veux pas dire "That we came up with at Kumo" plutôt ?

Reply:
`Instead of computing the same aggregate in every projector` tu n'as jamais parlé de ça avant.

Reply:
Yes c'est mieux t'as raison

Reply:
Pas faux, ça fait bizarre comme formulation du coup